Task 204 documentation

lindsayrblank opened the pull request:
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>
Issue #204 Updated and added METviewer documentation.

- [X] Recommend testing for the reviewer to perform, including the location of input datasets:</br>
Read the METviewer documentation (I will email all requested reviewers and we can split up who reads what). 

- [X] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
No, it's documentation.

- [X] After merging, should the reviewer **DELETE** the feature branch from GitHub? **[Yes or No]**</br>
No, I'll do that. 

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.

Commit:
Updated documentation

Commit:
Updated seriesplots instructions

Commit:
Added xml to series plot chapter

Commit:
Bar plot chapter

Commit:
Box plot chapter

Commit:
ROC Plot Chapter

Commit:
Reliability diagram chapter

Commit:
Performance diagram chapter

Commit:
Contour plot chapter

Commit:
Spread/skill chapter

Commit:
ECLV draft: in progress

Commit:
Taylor diagram and updated chapters

Commit:
ECLV chapter and line type sections in other chapters

Commit:
Finished scorecard chapter with examples

Commit:
Finished line type section for Scorecard chapter

jprestop committed:
Update seriesplots.rst

Corrected spelling for approximately.

jprestop committed:
Update scorecard.rst

Added a period after the following sentence and italicized "include/": "The folder that will contain the data and generated R scripts, must contain the sub-folder include/."

jprestop committed:
Update scorecard.rst

Italicized R_tmpl/

jprestop committed:
Update scorecard.rst

Added "an" to the sentence "This is optional element."

jprestop committed:
Update eclvplots.rst

Made some minor grammar changes.

jprestop commented on docs/scorecard.rst:
```diff
@@ -1,136 +1,188 @@
+
+**<folders>:** A structure that specifies the input and output file system folders.
+        
+        **<r_tmpl>:** The folder that contains the R templates used by METviewer, typically *R_tmpl/* from the metviewer CVS source tree.
```
```suggestion
        **<r_tmpl>:** The folder that contains the R templates used by METviewer, typically *R_tmpl/* from the METviewer CVS source tree.
```
Lindsay, should METviewer be capitalized here? I am not sure, but thought I should ask.

Commit:
Update docs/scorecard.rst

Change capitalization

Co-authored-by: jprestop <jpresto@ucar.edu>

jprestop committed:
Update scorecard.rst

Adding ?? around the sentence that we need to follow up on later.

jprestop committed:
Update eclvplots.rst

Adding ?? around sentence to insert link later so that we can find it more easily.

jprestop committed:
Update spreadskillplot.rst

Made some minor grammar changes, corrected link, and added "??" around text that needs follow up for easier finding.

jprestop committed:
Update contourplot.rst

Minor grammar change and added "??" where follow up is needed for easier recognition.

jprestop committed:
Update taylordiag.rst

Made minor grammar changes and added "??" to what needs to be revisited for quick reference.

jprestop committed:
Update spreadskillplot.rst

Fixed spelling of "approximately".

jprestop committed:
Update eclvplots.rst

Fixed spelling of "approximately".

jprestop committed:
Update perfdiag.rst

Made minor grammar changes and added "??" around a part that needs to be revisited.

jprestop committed:
Update reliabilityplots.rst

Made minor grammar changes and put "??" around part to be revisited.

jprestop committed:
Update rocplot.rst

Made minor grammar changes, fixed link to the MET User's Guide, and added "??" around text that needs to be revisited.

jprestop committed:
Update reliabilityplots.rst

Added "that" to a sentence.

jprestop committed:
Update perfdiag.rst

Added "that" to a sentence.

jprestop committed:
Update taylordiag.rst

Added "that" to a sentence.

jprestop committed:
Update contourplot.rst

jprestop committed:
Update spreadskillplot.rst

jprestop committed:
Update eclvplots.rst

jprestop committed:
Update boxplot.rst

Made minor grammar changes, fixed the link to the MET User's Guide, and added "??" around text to be revisited in order to find it more easily.

jprestop committed:
Update barplots.rst

Made minor grammar changes, fixed the link to the MET User's Guide, and added "??" to text to be revisited later.

jprestop committed:
Update boxplot.rst

Correct spelling

jprestop committed:
Update perfdiag.rst

Corrected spelling

jprestop committed:
Update contourplot.rst

Corrected spelling

jprestop committed:
Update spreadskillplot.rst

Corrected spelling

jprestop committed:
Update eclvplots.rst

Corrected spelling

jprestop committed:
Update seriesplots.rst

Minor grammar changes, fixed link to MET User's Guide, added "??" around section to be revisited.

jprestop reviewed (commented):
Chapters 3 - 13 have been reviewed for grammar, links, etc.  They are now ready for Tara's and Tatiana's review. Please note that "??" have been added around sections that needed to be revisited.  For example "??[insert link to XML]??" and "??NOTE: This is in Tatiana’s HTML documentation we’ll need to incorporate??" so that we can run "grep" to find these places more easily once we're ready to update them, so please do not remove these "??".  Thanks!

jprestop commented:
I forgot to mention that I uploaded the changes I made to this site:https://ral.ucar.edu/staff/jpresto/work/METviewer_jpresto/index.html

So you'll want to look here rather than in Lindsay's area because she does not have the changes from the review in her area.

jprestop committed:
Committing changes to index.rst from develop

Commit:
Fixed conflicts between develop and task_204 for index.rst

Commit:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
creating new file directory structure, updating locations in rst files, changing png file permissions and changing commas in references.

jprestop committed:
Removed files from this area as they are now in the Users_Guide subdirectory

jprestop committed:
Adding figure and xml directories

jprestop committed:
Updating index.rst file

Commit:
Tranisitioning HTML documentation to sphinx

TatianaBurek commented:
The example of the Reliability plot was build with a bug that we fixed in beta2 - No-Skill line is above the Perfect reliability line. We should change this example with something that doesn't have this bug

TatianaBurek committed:
changed METVIEWER to MV_HOME #208

TatianaBurek committed:
Removed 'Revision statistics', a few corrections #208

TatianaBurek committed:
change FHO line type to NBRCTC #208

TatianaBurek committed:
removed CNT line type #208

TatianaBurek commented:
METviewer installation guide is out of date and needs to be updated

Commit:
Fully formatted batch.rst

Commit:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

Commit:
Fully formatted database scrubbing chapter

Commit:
Fully formatted testing module chapter

Commit:
Fully formatted web service chapter

Commit:
Full formatted database_loading chapter

Commit:
Incldued release notes in Overview Chapter

Commit:
Table added to Database Loading chapter

Commit:
Updated images and tables

TatianaBurek committed:
new example of Rely plot

jprestop committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

jprestop committed:
Added met_viewer_logo_2019_09.png to docs/_static

jprestop committed:
Added METplus_banner_photo_web.png to docs/_static

jprestop committed:
Made minor modifications

TatianaBurek committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

jprestop commented:
Tatiana, with regard to "METviewer installation guide is out of date and needs to be updated" and "The example of the Reliability plot was build with a bug that we fixed in beta2 - No-Skill line is above the Perfect reliability line. We should change this example with something that doesn't have this bug" - do these still need to be done or have they already been done?

TatianaBurek commented:
I changed the Reliability plot example already but the installation guide will be updated later. I have a corresponding issue in this repo

jprestop commented:
Awesome! Thank you for taking care of all of this!  Are there chapters that Tara is scheduled to cover or did you cover everything?

jprestop committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

TatianaBurek committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

TatianaBurek committed:
new AF features in scorecard #208

lisagoodrich committed:
fixing typos, web links and such.

jprestop commented on docs/Users_Guide/overview.rst:
```diff
@@ -0,0 +1,277 @@
+initialization, same region, etc…).  Use of this method is recommended when
+trying to compare the skill of one or more predictions.
+To test, I tried method (1) and it works fine for some of the categorical
+stats, like BASER and FMEAN. However, applying it to Gilbert Skill Score  (GSS) results in this error:
```
> To test, I tried method (1) and it works fine for some of the categorical
stats, like BASER and FMEAN. However, applying it to Gilbert Skill Score  (GSS) results in this error:

Hi Lindsay.  This line of text refers to "this error", with a colon after it as if an error were going to be beneath it, but there is no error.  Do you have an error to add here?  If not, could you please reword it to say "results in an error." or something like that?  Please do a "git pull" before starting on any changes, as I know Lisa committed some changes to this file.  Thanks!

TatianaBurek commented:
All, 
I'm planning to release METviewer-beta2 today or tomorrow. If we want to include this PR into beta2 we need to finish it and merge into develop today.

Tatiana

lindsayrblank commented:
Hi all,

I actually don't know what is supposed to go after "this error". That was
from Tressa, and I can't find any reference to it in Tatiana's
documentation. Tatiana, do you know what could go there? If not, we can get
rid of it.

Thanks,
Lindsay

On Mon, Dec 7, 2020 at 12:59 PM Tatiana Burek <notifications@github.com>
wrote:

> All,
> I'm planning to release METviewer-beta2 today or tomorrow. If we want to
> include this PR into beta2 we need to finish it and merge into develop
> today.
>
> Tatiana
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METviewer/pull/208#issuecomment-740146166>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABR7BBU4BUB77PO4BWJH7PLSTUXYRANCNFSM4STYZUSA>
> .
>


-- 
Lindsay R. Blank
Associate Scientist
Joint Numerical Testbed Program
National Center for Atmospheric Research
+1 (303)-497-2772

*My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.*

TatianaBurek commented:
I don't know how  and why Tressa got the error. GSS stat should work in EE without errors .

Tatiana

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lindsayrblank commented:
Alright, then I'll delete that line and push the changes. Thanks!

On Mon, Dec 7, 2020 at 2:22 PM Tatiana Burek <notifications@github.com>
wrote:

> I don't know how and why Tressa got the error. GSS stat should work in EE
> without errors .
>
> Tatiana
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METviewer/pull/208#issuecomment-740187641>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABR7BBTX7EMF7WWQXTEGFQ3STVBRZANCNFSM4STYZUSA>
> .
>


-- 
Lindsay R. Blank
Associate Scientist
Joint Numerical Testbed Program
National Center for Atmospheric Research
+1 (303)-497-2772

*My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.*

Commit:
Removed line with incorrect EE information

TatianaBurek committed:
updated installation instructions

TatianaBurek committed:
updated installation instructions

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

TatianaBurek committed:
Update install instructions #221 #208

lisagoodrich committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
updating formatting.

jprestop committed:
Formatting changes

lisagoodrich committed:
fixing formatting

lisagoodrich committed:
fixing conflict

lisagoodrich committed:
updating formatting

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
fixing formatting, typos, etc.

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
unclear if any change happened

jprestop committed:
Reworded a paragraph and other grammar and formatting changes

lisagoodrich committed:
updating formatting

lisagoodrich committed:
removing parentheses

lisagoodrich committed:
fixing parenthesis

TatianaBurek committed:
documented the fact that at least one <load_val> entry is required

TatianaBurek committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
fixing small typos

lisagoodrich committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

jprestop committed:
Moved docs/Users_Guide/xml to _static/xml

jprestop committed:
Updated links to xml files

jprestop committed:
Added link to test xml file linking

jprestop committed:
Fixed link language.

lisagoodrich committed:
adding references

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
adding xml web links

lisagoodrich committed:
updating settings so figures automatically number

lisagoodrich committed:
updating formatting/typos

lisagoodrich committed:
fixing formatting/typos etc.

lisagoodrich committed:
updating references

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

jprestop committed:
Update references.rst

Added extra "| " at the end to add a line break.

lisagoodrich committed:
updating typos, references, formatting

lisagoodrich committed:
making updates

jprestop committed:
Fixing up formatting

TatianaBurek committed:
Scorecard must have  at least 3 levels of <field> elements

TatianaBurek committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

TatianaBurek committed:
add comment about scorecard SINGLE mode

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
updating references

lisagoodrich committed:
updating figure numbering

TatianaBurek committed:
remove 'met_version' tag from the loading documentation #232

lisagoodrich committed:
adding return lines for ease of reading

lisagoodrich committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
Merge branch 'task_204_documentation' of https://github.com/dtcenter/METviewer into task_204_documentation

lisagoodrich committed:
Changing from 2nd person to 3rd person

lisagoodrich committed:
removing quotations for continunity

lisagoodrich committed:
adding html link

jprestop committed:
Setting up Contributors_Guide to be ignored for now

jprestop committed:
Adding Contributor's Guide and moving testing.rst from User's Guide to Contributor's Guide

jprestop committed:
Removed testing from index.rst as testing.rst now lives in the Contributor's Guide.

jprestop committed:
Removed Contributors_Guide as it does not show up in the menu if it is not added to the docs/index.rst file

jprestop committed:
Merge branch 'develop' of https://github.com/dtcenter/METviewer into task_204_documentation

jprestop reviewed (approved):
I have reviewed these changes and they look good.  Thank you all for your hard work on this task!

jprestop commented:
I noticed that the base of this pull request was main_v3.1, but it should be "develop" so I updated the base so that these modification would get merged into "develop".
